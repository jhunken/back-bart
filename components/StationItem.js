import React, { Component } from 'react'
import { StyleSheet } from 'react-native'
import { ListItem, Text } from 'react-native-elements'

const StationItem = (props) => {
  let item = props.station
  let getSubtitle = (station) => {
    if (!station.key) {
      return ''
    }
    let destETDMinsStr
    let backETDMinsStr

    if (station && station.destETD && station.destETD.minutes) {
      destETDMinsStr = station.destETD.minutes
      if (destETDMinsStr !== 'Leaving') {
        destETDMinsStr = destETDMinsStr + 'm'
      }
    }

    if (station.backETD && station.backETD.minutes) {
      backETDMinsStr = station.backETD.minutes
      if (backETDMinsStr !== 'Leaving') {
        backETDMinsStr = backETDMinsStr + 'm'
      }
    }

    return `▼ Dest: ${destETDMinsStr || 'n/a'}  ▲ Back: ${backETDMinsStr || 'n/a'}`
  }

  if (item.hasETD) {
    return (
      <ListItem
        roundAvatar
        title={(<Text style={styles.title}>{item.key}</Text>)}
        subtitle={getSubtitle(item)}
        containerStyle={{borderBottomWidth: 0}}
        onPress={() => props.selectStation(item)}
        underlayColor='#E0F2F1'
        style={(props.departureStation && props.departureStation.abbr && item.abbr.toLocaleLowerCase() === props.departureStation.abbr.toLowerCase()) ? styles.selected : {}}
      />)
  }
  return null
}

export const styles = StyleSheet.create({
  selected: {
    backgroundColor: '#B2DFDB',
  },
  title: {
    fontSize: 16,
    fontWeight: 'bold',
    paddingTop: 4,
    paddingBottom: 4,

  },
})

export default StationItem