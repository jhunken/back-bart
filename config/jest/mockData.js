export const stations = [
  {
    name: 'station 1',
    abbr: 'stn1',
    hasETD: false
  },
  {
    'name': 'Warm Springs/South Fremont',
    'abbr': 'WARM',
    'etd': [{
      'destination': 'Daly City',
      'abbreviation': 'DALY',
      'limited': '0',
      'estimate': [{
        'minutes': 'Leaving',
        'platform': '2',
        'direction': 'South',
        'length': '10',
        'color': 'GREEN',
        'hexcolor': '#339933',
        'bikeflag': '1',
        'delay': '99'
      }, {
        'minutes': '22',
        'platform': '2',
        'direction': 'South',
        'length': '10',
        'color': 'GREEN',
        'hexcolor': '#339933',
        'bikeflag': '1',
        'delay': '0'
      }, {
        'minutes': '37',
        'platform': '2',
        'direction': 'South',
        'length': '10',
        'color': 'GREEN',
        'hexcolor': '#339933',
        'bikeflag': '1',
        'delay': '0'
      }]
    }],
    'key': 'WARM',
    'hasETD': true,
    'destETD': {
      'minutes': 'Leaving',
      'platform': '2',
      'direction': 'South',
      'length': '10',
      'color': 'GREEN',
      'hexcolor': '#339933',
      'bikeflag': '1',
      'delay': '99'
    },
    'backETD': {'minutes': null}
  }
]