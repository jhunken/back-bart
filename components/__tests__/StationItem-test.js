import React from 'react'
import 'react-native'

import renderer from 'react-test-renderer'

import StationItem from '../StationItem'
import { stations } from '../../config/jest/mockData'

test('renders an empty station correctly', () => {
  const tree = renderer.create(
    <StationItem
      station={stations[0]}
      selectStation={jest.fn()}
      departureStation={jest.fn()}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders a valid station correctly', () => {
  const tree = renderer.create(
    <StationItem
      station={stations[1]}
      selectStation={jest.fn()}
      departureStation={jest.fn()}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})