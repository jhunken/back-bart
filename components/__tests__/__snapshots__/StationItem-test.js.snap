// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a valid station correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {},
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginLeft": 10,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={undefined}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingBottom": 4,
                  "paddingTop": 4,
                },
              ]
            }
          >
            WARM
          </Text>
        </View>
      </View>
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#86939e",
                  "fontSize": 15,
                  "fontWeight": "600",
                  "marginTop": 1,
                },
                Object {
                  "marginLeft": 10,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          ▼ Dest: Leaving  ▲ Back: n/a
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 0.15,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        onPress={undefined}
        style={
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        underlayColor="white"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#bdc6cf",
                "fontSize": 28,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders an empty station correctly 1`] = `null`;
